common-pipeline/pipeline-flow/pipeline-flow-v3-schema.json

```diff
@@ -340,7 +340,8 @@
       },
       "required": [
         "id",
-        "type"
+        "type",
+        "outputs"
```

Comment:
Are outputs actually required?

Reply:
Yes, that is the idea behind the issue. The problem is that binding entry nodes and binding exit nodes are identical except that the former has `outputs` and latter has `inputs. Both have the same type which unfortunately is just 'binding'.  

As an aside: I think this was a mistake when the schema was first created because since entry and exit nodes are different, even though only slightly, they should have been given their own type names.   

Anyway, since the current schema says that `binding` nodes only 'required fields' are `id` and `type` that means if we have a node where, say, id = "1234" and type = "binding", we cannot know whether it is a binding input or binding output node. But the schema contains the following (where `oneOf` means one and only one) : 
```
"node_type_def": {
      "oneOf": [
        {
          "$ref": "#/definitions/execution_node_def"
        },
        {
          "$ref": "#/definitions/supernode_def"
        },
        {
          "$ref": "#/definitions/binding_entry_node_def"
        },
        {
          "$ref": "#/definitions/binding_exit_node_def"
        },
        {
          "$ref": "#/definitions/model_node_def"
        }
      ]
    },
 ```
This means we would get a validation error with such a node because even though it conforms to the required fields, in that it has an `id` and `type`, the validator throws a validation exception because it is unable to figure out if the node is one  of the five definitions specified in the `oneOf` construct.

Now, since the only differentiating factor between binding entry and binding exit nodes is the existence of either the `outputs` or `inputs` field then that means those nodes MUST have either an outputs field or an inputs field for validation to pass.

So extending the required fields, as done in this PR, with `outputs` and `inputs` is not changing the definition but just reinforcing what is necessary to get the validation to pass.

This issue came up in the early days when the Data Stage team were playing with pipelineFlow documents and they created a node with an `id` and `type = "binding"` and didn't understand the error message from the schema validator.

Reply:
Makes sense.  Thanks for explaining it.